p2p/stream/types/stream.go

```diff
@@ -59,20 +67,64 @@ func (st *BaseStream) ProtoSpec() (ProtoSpec, error) {
 
 // Close close the stream on both sides.
 func (st *BaseStream) Close() error {
-	return st.raw.Reset()
+	return st.raw.Close()
 }
 
-// WriteBytes write the bytes to the stream
+const (
+	maxMsgBytes = 20 * 1024 * 1024 // 20MB
```

Comment:
have you tested the performance of max 20MB data in streaming? Do we really need to support this huge size?

Reply:
This is currently the theoretical upper limit. As I said in description, for stream protocol, message is capped at 10 blocks, which implies there is at most 20MB data. For real cases, it's less than 200KB, based on the block size.